addons/markinglaser/functions/fnc_renderPFH.sqf

```diff
@@ -0,0 +1,63 @@
+#include "script_component.hpp"
+/*
+ * Author: BaerMitUmlaut
+ * Renders all marking lasers.
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [] call ace_markinglaser_fnc_renderPFH
+ *
+ * Public: No
+ */
+
+if (GVAR(lasers) isEqualTo []) exitWith {};
+
+#ifndef DEBUG_MODE_FULL
+if (!IN_NVGS) exitWith {
+    { _x hideObject true } forEach GVAR(lasers);
+};
+#endif
+
+{
+    private _laser = _x;
+    private _aircraft = attachedTo _laser;
+    private _origin = _aircraft getVariable [QGVAR(laserOrigin), [0, 0, 0]];
+    private _originWorld = _aircraft modelToWorldVisualWorld _origin;
+
+    // Unhide laser in case NVGs were turned off
+    _laser hideObject false;
+
+    // Update laser direction
+    private _vector = [_aircraft, _laser] call FUNC(getVector);
+    _laser setVectorUp (_aircraft vectorWorldToModelVisual _vector);
+
+    // Adjust length of laser so it doesn't go through objects
+    private _intersectPos = lineIntersectsSurfaces [
+        _originWorld,
+        _originWorld vectorAdd (_vector vectorMultiply LIS_LIMIT),
+        _laser
+    ] param [0, []] param [0, []];
+
+    if (_intersectPos isEqualTo []) then {
+        _laser animate ["adjustLength", 0, true];
```

Comment:
wiki says to use "animateSource" instead

Reply:
Both are Effect Global however the PFH seems to be supposed to be running local?